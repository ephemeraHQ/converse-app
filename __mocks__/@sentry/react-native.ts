const init = jest.fn();
const captureException = jest.fn();
const captureMessage = jest.fn();
const addBreadcrumb = jest.fn();
const withScope = jest.fn();

const MockedSentry = {
  init,
  captureException,
  captureMessage,
  addBreadcrumb,
  withScope,
};

export { init, captureException, captureMessage, addBreadcrumb, withScope };

export default MockedSentry;
