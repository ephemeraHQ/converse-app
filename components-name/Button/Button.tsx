import { Button } from "../../design-system/Button/Button";

/**
 * @deprecated This component is deprecated. Please use the Button component from the design system directly.
 * Import it from "../../design-system/Button/Button" instead.
 */
export default Button;
