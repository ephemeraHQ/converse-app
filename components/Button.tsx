import React from "react";
import {
  ColorSchemeName,
  GestureResponderEvent,
  PlatformColor,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  useColorScheme,
  ViewStyle,
} from "react-native";
import { SFSymbol } from "react-native-sfsymbols";

import { buttonGreyBackground } from "../utils/colors";

type Props = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  variant: "blue" | "grey" | "text";
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  allowFontScaling?: boolean;
  picto?: string;
};

export default function Button({
  title,
  onPress,
  variant,
  style,
  textStyle,
  picto,
  allowFontScaling = true,
}: Props) {
  const colorScheme = useColorScheme();
  const styles = getStyles(colorScheme);
  const buttonStyle =
    variant === "blue"
      ? styles.buttonBlue
      : variant === "grey"
      ? styles.buttonGrey
      : styles.buttonText;
  const buttonTextStyle =
    variant === "blue"
      ? styles.buttonBlueText
      : variant === "grey"
      ? styles.buttonGreyText
      : styles.buttonTextText;
  return (
    <TouchableOpacity style={[buttonStyle, style]} onPress={onPress}>
      {picto && (
        <SFSymbol
          name={picto}
          weight="bold"
          scale="large"
          color="white"
          size={13}
          resizeMode="center"
          multicolor={false}
          style={styles.picto}
        />
      )}
      <Text
        style={[buttonTextStyle, textStyle]}
        allowFontScaling={allowFontScaling}
      >
        {title}
      </Text>
    </TouchableOpacity>
  );
}

const getStyles = (colorScheme: ColorSchemeName) =>
  StyleSheet.create({
    buttonBlue: {
      backgroundColor: PlatformColor("systemBlue"),
      display: "flex",
      alignSelf: "stretch",
      marginHorizontal: 32,
      textAlign: "center",
      paddingVertical: 14,
      borderRadius: 14,
      flexDirection: "row",
      justifyContent: "center",
    },
    buttonBlueText: {
      color: "white",
      textAlign: "center",
      fontWeight: "600",
      fontSize: 17,
    },
    buttonGrey: {
      backgroundColor: buttonGreyBackground(colorScheme),
      paddingHorizontal: 15,
      paddingVertical: 4,
      borderRadius: 100,
    },
    buttonGreyText: {
      color: PlatformColor("systemBlue"),
      textAlign: "center",
      fontWeight: "600",
      fontSize: 17,
    },
    buttonText: {},
    buttonTextText: {
      color: PlatformColor("systemBlue"),
      textAlign: "center",
      fontWeight: "400",
      fontSize: 17,
    },
    picto: {
      marginRight: 15,
    },
  });
