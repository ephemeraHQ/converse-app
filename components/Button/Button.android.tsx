import React from "react";
import {
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  TextStyle,
  View,
  Text,
  ViewStyle,
} from "react-native";
import { Button as MaterialButton } from "react-native-paper";

import Picto from "../Picto/Picto.android";

type Props = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  variant: "primary" | "secondary" | "grey" | "text";
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  allowFontScaling?: boolean;
  picto?: string;
};

export default function Button({
  title,
  onPress,
  variant,
  style,
  textStyle,
  picto,
  allowFontScaling = true,
}: Props) {
  return (
    <View
      style={[
        variant === "primary"
          ? styles.buttonPrimaryContainer
          : variant === "secondary"
          ? styles.buttonSecondaryContainer
          : undefined,
        style,
      ]}
    >
      <MaterialButton
        mode={
          variant === "primary" || variant === "secondary"
            ? "contained"
            : "text"
        }
        style={
          variant === "primary" || variant === "secondary"
            ? styles.buttonPrimary
            : undefined
        }
        onPress={onPress}
        icon={
          picto
            ? ({ color, size }) => (
                <Picto
                  picto={picto}
                  color={color}
                  size={size}
                  style={variant === "secondary" ? undefined : styles.picto}
                />
              )
            : undefined
        }
      >
        <Text style={textStyle}>{title}</Text>
      </MaterialButton>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonPrimaryContainer: {
    paddingHorizontal: 32,
    width: "100%",
  },
  buttonSecondaryContainer: {
    paddingHorizontal: 16,
    width: "auto",
  },
  buttonPrimary: {
    width: "100%",
  },
  picto: {
    marginRight: 10,
  },
});
