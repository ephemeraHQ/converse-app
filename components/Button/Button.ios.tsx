import {
  dangerColor,
  inversePrimaryColor,
  primaryColor,
  tertiaryBackgroundColor,
} from "@styles/colors";
import { PictoSizes } from "@styles/sizes";
import React from "react";
import {
  GestureResponderEvent,
  Platform,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  useColorScheme,
  ViewStyle,
} from "react-native";

import Picto from "../Picto/Picto";

type Props = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  variant: "primary" | "secondary" | "secondary-danger" | "grey" | "text";
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  allowFontScaling?: boolean;
  picto?: string;
  numberOfLines?: number;
  hitSlop?: number;
};

export default function Button({
  title,
  onPress,
  variant,
  style,
  textStyle,
  picto,
  allowFontScaling = true,
  numberOfLines,
  hitSlop,
}: Props) {
  const colorScheme = useColorScheme();
  const styles = useStyles();
  const buttonStyle =
    variant === "primary"
      ? [styles.buttonPrimary]
      : variant === "secondary"
      ? [styles.buttonPrimary, styles.buttonSecondary]
      : variant === "grey"
      ? [styles.buttonGrey]
      : variant === "secondary-danger"
      ? [styles.buttonPrimary, styles.buttonSecondary, styles.buttonDanger]
      : [styles.buttonText];
  const buttonTextStyle =
    variant === "primary"
      ? [styles.buttonPrimaryText]
      : variant === "secondary" || variant === "secondary-danger"
      ? [styles.buttonPrimaryText, styles.buttonSecondaryText]
      : variant === "grey"
      ? [styles.buttonGreyText]
      : [styles.buttonTextText];
  return (
    <TouchableOpacity
      style={[...buttonStyle, style]}
      onPress={onPress}
      hitSlop={hitSlop}
    >
      {picto && (
        <Picto
          picto={picto}
          size={variant === "text" ? PictoSizes.textButton : PictoSizes.button}
          style={[
            styles.picto,
            variant === "text" ? { paddingLeft: 15 } : undefined,
          ]}
          color={
            variant === "text"
              ? primaryColor(colorScheme)
              : inversePrimaryColor(colorScheme)
          }
        />
      )}
      <Text
        style={[...buttonTextStyle, textStyle]}
        allowFontScaling={allowFontScaling}
        numberOfLines={numberOfLines}
      >
        {title}
      </Text>
    </TouchableOpacity>
  );
}

const useStyles = () => {
  const colorScheme = useColorScheme();
  return StyleSheet.create({
    buttonPrimary: {
      backgroundColor: primaryColor(colorScheme),
      display: "flex",
      alignSelf: "stretch",
      marginHorizontal: 32,
      textAlign: "center",
      paddingVertical: 14,
      borderRadius: 14,
      flexDirection: "row",
      justifyContent: "center",
    },
    buttonPrimaryText: {
      color: inversePrimaryColor(colorScheme),
      textAlign: "center",
      fontWeight: "600",
      fontSize: 17,
    },
    buttonSecondary: {
      paddingRight: 15,
      paddingLeft: 15,
      borderRadius: 100,
      paddingVertical: 7,
    },
    buttonDanger: {
      backgroundColor: dangerColor(colorScheme),
    },
    buttonSecondaryText: {
      fontWeight: "400",
    },
    buttonGrey: {
      backgroundColor: tertiaryBackgroundColor(colorScheme),
      paddingHorizontal: 15,
      paddingVertical: 4,
      borderRadius: 100,
    },
    buttonGreyText: {
      color: colorScheme === "dark" ? "white" : primaryColor(colorScheme),
      textAlign: "center",
      fontWeight: "600",
      fontSize: 17,
    },
    buttonText: {
      flexDirection: "row",
    },
    buttonTextText: {
      color: Platform.OS === "ios" ? primaryColor(colorScheme) : "blue",
      textAlign: "center",
      fontWeight: "400",
      fontSize: 17,
    },
    picto: {
      marginRight: 15,
      marginLeft: 10,
    },
  });
};
