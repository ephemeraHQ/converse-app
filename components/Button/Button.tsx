import React from "react";
import {
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";
import { Button as MaterialButton } from "react-native-paper";

import Picto from "../Picto/Picto";

type Props = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  variant: "primary" | "secondary" | "grey" | "text" | "secondary-danger";
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  allowFontScaling?: boolean;
  picto?: string;
};

export default function Button({
  title,
  onPress,
  variant,
  style,
  textStyle,
  picto,
  allowFontScaling = true,
}: Props) {
  return (
    <View
      style={[
        variant === "primary"
          ? styles.buttonPrimaryContainer
          : variant === "secondary"
          ? styles.buttonSecondaryContainer
          : undefined,
        style,
      ]}
    >
      <MaterialButton
        mode={
          variant === "primary" || variant === "secondary"
            ? "contained"
            : "text"
        }
        style={
          variant === "primary" || variant === "secondary"
            ? styles.buttonPrimary
            : undefined
        }
        onPress={onPress}
        labelStyle={textStyle}
        icon={
          picto
            ? ({ color, size }) => (
                <Picto
                  picto={picto}
                  color={color}
                  size={size}
                  style={
                    variant === "secondary" || variant === "text"
                      ? undefined
                      : styles.picto
                  }
                />
              )
            : undefined
        }
      >
        <Text style={styles.textContainer}>{title}</Text>
      </MaterialButton>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonPrimaryContainer: {
    paddingHorizontal: 32,
    width: "100%",
  },
  buttonSecondaryContainer: {
    paddingHorizontal: 16,
    width: "auto",
  },
  buttonPrimary: {
    width: "100%",
  },
  picto: {
    marginRight: 10,
  },
  textContainer: {
    flexGrow: 1,
  },
});
