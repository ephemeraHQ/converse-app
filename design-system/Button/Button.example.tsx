import { VStack } from "@design-system/VStack";
import { useAppTheme } from "@theme/useAppTheme";
import { Button } from "./Button";

type IExampleProps = {
  onPress?: () => void;
};

export function ButtonExample(args: IExampleProps) {
  const { onPress } = args;

  const { theme } = useAppTheme();

  return (
    <VStack
      style={{
        gap: theme.spacing.md,
        backgroundColor: theme.colors.background.raised,
      }}
    >
      {/* Basic Variants */}
      <Button text="Fill Button" variant="fill" onPress={onPress} />
      <Button text="Outline Button" variant="outline" onPress={onPress} />
      <Button text="Link Button" variant="link" onPress={onPress} />

      {/* Sizes */}
      <Button text="Small Button" size="sm" onPress={onPress} />
      {/* <Button text="Medium Button" size="md" onPress={onPress} /> */}
      <Button text="Large Button" size="lg" onPress={onPress} />

      {/* States */}
      <Button text="Loading Button" loading={true} onPress={onPress} />
      {/* <Button text="Disabled Button" disabled={true} onPress={onPress} /> */}

      {/* With Icons */}
      <Button text="Button with Icon" icon="search" onPress={onPress} />

      {/* Custom Accessories */}
      {/* <Button
        size="sm"
        text="Custom Left Accessory"
        LeftAccessory={({ style }) => (
          <Icon icon="chevron.left" color="white" style={style} />
        )}
        onPress={onPress}
      /> */}

      {/* i18n Example */}
      {/* <Button tx="cancel" txOptions={{ name: "John" }} onPress={onPress} /> */}

      {/* With Haptic Feedback */}
      <Button text="Haptic Button" withHapticFeedback onPress={onPress} />
    </VStack>
  );
}
