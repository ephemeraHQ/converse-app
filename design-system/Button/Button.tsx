import { useAppTheme } from "@theme/useAppTheme";
import { useCallback } from "react";
import {
  Pressable,
  PressableStateCallbackType,
  StyleProp,
  TextStyle,
  ViewStyle,
} from "react-native";

import Picto from "../../components/Picto/Picto";
import { Text } from "../Text";
import { IButtonProps, IButtonVariant } from "./Button.props";
import {
  $buttonLeftAccessoryStyle,
  $buttonRightAccessoryStyle,
  getButtonTextStyle,
  getButtonViewStyle,
} from "./Button.styles";

export function Button(props: IButtonProps) {
  const {
    tx,
    text,
    txOptions,
    style: $viewStyleOverride,
    pressedStyle: $pressedViewStyleOverride,
    textStyle: $textStyleOverride,
    pressedTextStyle: $pressedTextStyleOverride,
    disabledTextStyle: $disabledTextStyleOverride,
    children,
    RightAccessory,
    LeftAccessory,
    disabled,
    disabledStyle: $disabledViewStyleOverride,
    size = "lg",
    // @deprecated,
    title,
    picto,
    ...rest
  } = props;

  const { themed, theme } = useAppTheme();

  const variant: IButtonVariant = props.variant ?? "fill";

  const $viewStyle = useCallback(
    ({ pressed }: PressableStateCallbackType): StyleProp<ViewStyle> => {
      return [
        themed(
          getButtonViewStyle({ variant, size, action: "primary", pressed })
        ),
        $viewStyleOverride,
        pressed && $pressedViewStyleOverride,
        disabled && $disabledViewStyleOverride,
      ];
    },
    [
      themed,
      variant,
      size,
      $viewStyleOverride,
      $pressedViewStyleOverride,
      $disabledViewStyleOverride,
      disabled,
    ]
  );

  const $textStyle = useCallback(
    ({ pressed }: PressableStateCallbackType): StyleProp<TextStyle> => {
      return [
        themed(
          getButtonTextStyle({ variant, size, action: "primary", pressed })
        ),
        $textStyleOverride,
        pressed && $pressedTextStyleOverride,
        disabled && $disabledTextStyleOverride,
      ];
    },
    [
      themed,
      variant,
      size,
      $textStyleOverride,
      $pressedTextStyleOverride,
      $disabledTextStyleOverride,
      disabled,
    ]
  );

  return (
    <Pressable
      style={$viewStyle}
      accessibilityRole="button"
      accessibilityState={{ disabled: !!disabled }}
      {...rest}
      disabled={disabled}
    >
      {(state) => (
        <>
          {!!LeftAccessory && (
            <LeftAccessory
              style={$buttonLeftAccessoryStyle}
              pressableState={state}
              disabled={disabled}
            />
          )}

          {/* @deprecated stuff */}
          {!!picto && (
            <Picto
              picto={picto}
              color={
                variant === "text" || variant === "link"
                  ? theme.colors.text.primary
                  : theme.colors.text.inverted.primary
              }
              style={themed($buttonLeftAccessoryStyle)}
            />
          )}

          <Text
            tx={tx}
            text={title ?? text}
            txOptions={txOptions}
            style={$textStyle(state)}
          >
            {children}
          </Text>

          {!!RightAccessory && (
            <RightAccessory
              style={$buttonRightAccessoryStyle}
              pressableState={state}
              disabled={disabled}
            />
          )}
        </>
      )}
    </Pressable>
  );
}
