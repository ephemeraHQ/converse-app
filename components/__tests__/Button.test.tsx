// __tests__/Button.test.tsx

import { render, fireEvent } from "@testing-library/react-native";
import React from "react";
import { Platform } from "react-native";

import Button from "../Button/Button";

// Mock useColorScheme for consistent testing
jest.mock("react-native/Libraries/Utilities/useColorScheme", () => ({
  __esModule: true,
  default: jest.fn(() => "light"),
}));

describe("Button Component", () => {
  describe("iOS Button", () => {
    it("renders correctly with primary variant", () => {
      Platform.OS = "ios"; // Set the platform to iOS

      const { toJSON, getByText } = render(
        <Button title="Primary Button" variant="primary" />
      );

      // Verify snapshot
      expect(toJSON()).toMatchSnapshot();

      // Check if the title is rendered
      expect(getByText("Primary Button")).toBeTruthy();
    });

    it("fires onPress event when clicked", () => {
      Platform.OS = "ios"; // Set the platform to iOS

      const onPressMock = jest.fn();
      const { getByText } = render(
        <Button
          title="Clickable Button"
          variant="primary"
          onPress={onPressMock}
        />
      );

      fireEvent.press(getByText("Clickable Button"));

      // Verify that the mock function is called
      expect(onPressMock).toHaveBeenCalled();
    });
  });

  describe("Default (Android) Button", () => {
    it("renders correctly with secondary-danger variant", () => {
      Platform.OS = "android"; // Set the platform to Android (or default)

      const { toJSON, getByText } = render(
        <Button title="Secondary Danger Button" variant="secondary-danger" />
      );

      // Verify snapshot
      expect(toJSON()).toMatchSnapshot();

      // Check if the title is rendered
      expect(getByText("Secondary Danger Button")).toBeTruthy();
    });

    it("renders correctly with picto", () => {
      Platform.OS = "android";

      const { toJSON, getByText } = render(
        <Button title="Button with Picto" variant="grey" picto="star" />
      );

      // Verify snapshot
      expect(toJSON()).toMatchSnapshot();

      // Check if the title is rendered
      expect(getByText("Button with Picto")).toBeTruthy();
    });
  });
});
