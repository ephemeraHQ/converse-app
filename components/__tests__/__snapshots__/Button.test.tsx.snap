// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Default (Android) Button renders correctly with picto 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#EFEFF0",
      "borderRadius": 100,
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 4,
    }
  }
>
  <RNSfsymbols
    iconColor={-1}
    multicolor={false}
    resizeMode="center"
    scale="large"
    size={13}
    style={
      [
        {
          "marginLeft": 10,
          "marginRight": 15,
        },
        undefined,
      ]
    }
    systemName="star"
    weight="regular"
  />
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "rgb(0,0,0)",
          "fontSize": 17,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Button with Picto
  </Text>
</View>
`;

exports[`Button Component Default (Android) Button renders correctly with secondary-danger variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "stretch",
      "backgroundColor": "rgb(186, 26, 26)",
      "borderRadius": 100,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 32,
      "opacity": 1,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingVertical": 7,
      "textAlign": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "rgb(255,255,255)",
          "fontSize": 17,
          "fontWeight": "600",
          "textAlign": "center",
        },
        {
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Secondary Danger Button
  </Text>
</View>
`;

exports[`Button Component iOS Button renders correctly with primary variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "stretch",
      "backgroundColor": "#000",
      "borderRadius": 14,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 32,
      "opacity": 1,
      "paddingVertical": 14,
      "textAlign": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFF",
          "fontSize": 17,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Primary Button
  </Text>
</View>
`;
